package/resource_monitor/resource_monitor/src/query_process_file_descriptor.c

```diff
@@ -0,0 +1,491 @@
+#endif
+  }
+  for (size_t i = 0; i < ITEM_COUNT && i < file_count; i++) {
+    state->top10_file_counts[i] = file_to_count[i].id;
+#ifdef DEBUG_QUERY_FD_TAB
+    PK_LOG_ANNO(LOG_DEBUG, "TOP 10 FD FILE : %s : %d", file_to_count[i].id, file_to_count[i].count);
+#endif
+  }
+}
+
+static bool query_fd_prepare(u16 *msg_type, u8 *len, u8 *sbp_buf, void *context)
+{
+  resq_state_t *state = context;
+
+  if (state->send_state == SEND_DONE) return false;
+
+  size_t index = (size_t)state->send_state;
+  switch (state->send_state) {
+  case SEND_FD_COUNTS_0:
+  case SEND_FD_COUNTS_1:
+  case SEND_FD_COUNTS_2:
+  case SEND_FD_COUNTS_3:
+  case SEND_FD_COUNTS_4:
+  case SEND_FD_COUNTS_5:
+  case SEND_FD_COUNTS_6:
+  case SEND_FD_COUNTS_7:
+  case SEND_FD_COUNTS_8:
+  case SEND_FD_COUNTS_9: {
+    pid_entry_t *entry = pid_table_get(state->pid_table, state->top10_pid_counts[index]);
+    *msg_type = SBP_MSG_LINUX_PROCESS_FD_COUNT;
+    msg_linux_process_fd_count_t *fd_counts = (msg_linux_process_fd_count_t *)sbp_buf;
+    fd_counts->index = (u8)index;
+    fd_counts->pid = entry->pid;
+    fd_counts->fd_count = entry->fd_count;
+#ifdef DEBUG_QUERY_FD_TAB
+    PK_LOG_ANNO(LOG_DEBUG,
+                "index : %d pid: %d : fd_count : %d\n",
+                fd_counts->index,
+                fd_counts->pid,
+                fd_counts->fd_count);
+#endif
+    *len = sizeof(msg_linux_process_fd_count_t);
+  } break;
+  case SEND_FD_SUMMARY:
+    *msg_type = SBP_MSG_LINUX_PROCESS_FD_SUMMARY;
+    msg_linux_process_fd_summary_t *fd_summary = (msg_linux_process_fd_summary_t *)sbp_buf;
+    fd_summary->sys_fd_count = state->total_fd_count;
+    int nBytes = 0;
+    for (int i = 0; i < ITEM_COUNT; i++) {
+
+      unsigned int buf_space =
+        (unsigned int)(SBP_FRAMING_MAX_PAYLOAD_SIZE - sizeof(msg_linux_process_fd_summary_t)
+                       - (unsigned int)nBytes);
+      file_entry_t *file_entry = file_table_get(state->file_table, state->top10_file_counts[i]);
+      nBytes += snprintf(&fd_summary->most_opened[nBytes],
+                         buf_space,
+                         "%d%c%s",
+                         file_entry->fd_count,
+                         '\0',
+                         file_entry->file_str);
+#ifdef DEBUG_QUERY_FD_TAB
+      PK_LOG_ANNO(LOG_DEBUG,
+                  "Summary:%d : %d %s bytes",
+                  nBytes,
+                  file_entry->fd_count,
+                  file_entry->file_str);
+#endif
+      if ((SBP_FRAMING_MAX_PAYLOAD_SIZE - sizeof(msg_linux_process_fd_summary_t) - 1)
+          < (unsigned int)nBytes) {
+        nBytes = (SBP_FRAMING_MAX_PAYLOAD_SIZE - sizeof(msg_linux_process_fd_summary_t) - 1);
+        break;
+      }
+    }
+    fd_summary->most_opened[nBytes] = '\0';
+    *len = (u8)(sizeof(msg_linux_process_fd_summary_t) + (unsigned)nBytes + 1);
+    break;
+  case SEND_DONE:
+  default: {
+#ifdef DEBUG_QUERY_FD_TAB
```

Comment:
Don't disable this error (unless there's a reason not to) it seems like a state should never happen so it should be safe to leave it enabled.